src/manage_nsfs/manage_nsfs_validations.js

```diff
@@ -441,25 +448,30 @@ async function validate_delete_account(config_root_backend, buckets_dir_path, ac
     });
 }
 
-    // TODO - when we have the structure of config we can check easily which IAM users are owned by the root account
-    // currently, partial copy from _list_config_files_for_users
-    async function check_if_root_account_does_not_have_IAM_users(config_root_backend, accounts_dir_path, account_to_check) {
-        const fs_context = native_fs_utils.get_process_fs_context(config_root_backend);
-        const entries = await nb_native().fs.readdir(fs_context, accounts_dir_path);
-        await P.map_with_concurrency(10, entries, async entry => {
-            if (entry.name.endsWith('.json')) {
-                const full_path = path.join(accounts_dir_path, entry.name);
-                const account_data = await get_config_data(config_root_backend, full_path);
-                if (entry.name.includes(config.NSFS_TEMP_CONF_DIR_NAME)) return undefined;
-                const is_root_account_owns_user = check_root_account_owns_user(account_to_check, account_data);
-                if (is_root_account_owns_user) {
-                        const detail_msg = `Account ${account_to_check.name} has IAM account ${account_data.name}`;
-                        throw_cli_error(ManageCLIError.AccountCannotBeRootAccountsManager, detail_msg);
-                }
-                return account_data;
+// TODO - when we have the structure of config we can check easily which IAM users are owned by the root account
+// currently, partial copy from _list_config_files_for_users
+/**
+ * @param {string} config_root_backend
+ * @param {string} accounts_dir_path
+ * @param {object} account_to_check
+ */
+async function check_if_root_account_does_not_have_IAM_users(config_root_backend, accounts_dir_path, account_to_check) {
```

Comment:
Why not passing `nsfs_config` instead of config_root_backend, accounts_dir_path? there are more places in which we still pass a lot of params instead of the global object?